text/0014-improve-locking-mechanism-for-parachains.md

```diff
@@ -0,0 +1,105 @@
+     2.2 Parachain never ever produced a block including previous slots.
+     2.3 A sufficiently long time has passed since the parachain is onboarded on a slot.
+
+Note that the lock status is only ignored when the condition is met. This means if the parachain produced a block after was considered stuck, the lock status will be enforced onwards.
+
+Furthermore, this is only considered as a short term solution. With planned coretime related changes, the whole slot mechanism will be replaced and this RFC will be obsolete.
+
+## Drawbacks
+
+Parachain locks are designed in such way to ensure the decentralization of parachains. This RFC proposes a way to bypass the lock mechanism and therefore could introduce centralization risk for new parachains.
+
+For example, one possible scenario is that a collectives may decide to launch a parachain together without centralized risk. However, if the parachain is unable to produce block, the parachain manager will be able to perform various actions without the consent of the collectives.
+
+It is considered this risk is tolerable as it requires the wasm/genesis to be invalid at first place or the parachain collators are fully controlled by centralized entity. It is not yet practically possible to develop a parachain without any centralized risk currently.
+
+Existing operational parachains will not be impacted due to condition 2.2.
+
+Long term impacts are not considered as this RFC is only a short term solution and will be obsolete once coretime related changes are implemented.
+
+## Testing, Security, and Privacy
+
+The implementation of this RFC will be tested on testnets (Rococo and Westend) first.
+
+An audit maybe required to ensure the implementation does not introduce unwanted side effects.
+
+There is no privacy related concerns.
+
+### Performance
+
+This RFC should not introduce any performance impact.
+
+### Ergonomics
+
+This RFC should improve the developer experiences for new and existing parachain teams
+
+### Compatibility
+
+This RFC is fully compatibility with existing interfaces.
+
+## Prior Art and References
+
+- Parachain Slot Extension Story: https://github.com/paritytech/polkadot/issues/4758
+- Allow parachain to renew lease without actually run another parachain: https://github.com/paritytech/polkadot/issues/6685
+- Always treat parachain that never produced block for a significant amount of time as unlocked: https://github.com/paritytech/polkadot/issues/7539
+
+## Unresolved Questions
+
+How long should we wait before considering a parachain is unable to produce block? It should not be too long otherwise will defects the purpose of this RFC. It should not be too short otherwise will introduce centralization risk. It is proposed to be 7 days.
+
+## Future Directions and Related Material
+
+Everything this RFC touches should be gone in the future once coretime related changes are implemented and therefore future considerations are not applicable.
```

Comment:
I do not believe this is true - while we will blur the distinction between parachains with and without slots, the `paras`, `paras_registrar`, and other mechanisms will remain, albeit with some alterations.

Reply:
We need a better generalized long term solution which will repalce this one.